### DISCUSSION TITLE
How to set that not save the picture of the verification result during the training process[Docs]

### DISCUSSION THREAD

**Discussion by 11923303233** [open-mmlab/mmediting#1975] [How-to] [answered]
### Branch

main branch  https://mmagic.readthedocs.io/en/latest/

### 📚 The doc issue

Hello，I want to speed up the training, how to set the validation without saving images.

**Comment by zengyh1900** [comment]
hi @11923303233 would you mind providing the config file you used?

**Comment by 11923303233** [comment]
> hi @11923303233 would you mind providing the config file you used?

I used configs/real_esrgan/realesrnet_c64b23g32_4xb12-lr2e-4-1000k_df2k-ost.py

**Comment by 11923303233** [comment]
train_cfg is IterBasedTrainLoop;
val_cfg is MultiValLoop

**Comment by zengyh1900** [comment, marked as answer]
hi @11923303233 
According to your provided config file, I think you are using `BasicVisualizationHook` to save images during validation, https://github.com/open-mmlab/mmagic/blob/main/configs/real_esrgan/realesrnet_c64b23g32_4xb12-lr2e-4-1000k_df2k-ost.py#L257C16-L257C38 
You can check more details about `BasicVisualizationHook` in our API documents https://mmagic.readthedocs.io/en/latest/autoapi/mmagic/engine/hooks/index.html#mmagic.engine.hooks.BasicVisualizationHook

To disable saving images during validation, you have several solutions, 
1. directly remove `BasicVisualizationHook` from your config, then you will not save images during both validation and test. 
2. disable saving images by setting augments as `dict(type='BasicVisualizationHook', interval=1, on_val=False)`
3. If you still want to save images during validation but with relatively fewer results, you can set the value of `interval`  as `dict(type='BasicVisualizationHook', interval=100)`

**Comment by 11923303233** [comment]
Trank you for your answer!  Previously, I only paid attention to other modules, not to BasicVisualizationHook
